To run a set of tests locally, I reverse-engineered what drone CI does:

- have an oC10 server installed locally
- from the root folder of the install (e.g. mine is /home/phil/git/owncloud/core ) clone the needed repos, create `user01` and `user02` and the expected calendars:
```
cd apps/dav/tests/ci/caldav
git clone https://github.com/apple/ccs-caldavtester.git CalDAVTester
git clone https://github.com/apple/ccs-pycalendar.git pycalendar
cd ../../../../../
OC_PASS=user01 php occ user:add --password-from-env user01
php occ dav:create-calendar user01 calendar
php occ dav:create-calendar user01 shared
OC_PASS=user02 php occ user:add --password-from-env user02
php occ dav:create-calendar user02 calendar
```

- run the PHP dev server in a terminal window of its own:
```
php -S 127.0.0.1:8889
```

- run the test suite:
```
cd apps/dav/tests/ci/caldav/CalDAVTester
PYTHONPATH="./pycalendar/src" python testcaldav.py --print-details-onfail --basedir "../../caldavtest/" -o cdt.txt \
	"CalDAV/current-user-principal.xml" \
	"CalDAV/sync-report.xml" \
	"CalDAV/sharing-calendars.xml"
```

I got mostly passes, 4 fails, so there is something in my local installation that is not quite the same as in CI (which is set up completely from scratch)

But this should be enough to help in doing a python3 refactoring - when I specify `python2` it runs, when I specify `python3` it falls over with:
```
Traceback (most recent call last):
  File "testcaldav.py", line 22, in <module>
    from src.manager import manager
  File "/home/phil/git/owncloud/core/apps/dav/tests/ci/caldav/CalDAVTester/src/manager.py", line 80
    print str
          ^
SyntaxError: Missing parentheses in call to 'print'. Did you mean print(str)?
```
That will be the first of perhaps many things that need to be changed.

--- a/apps/dav/tests/ci/caldav-old-endpoint/script.sh
+++ b/apps/dav/tests/ci/caldav-old-endpoint/script.sh
@@ -13,7 +13,7 @@ sleep 30
 
 # run the tests
 cd "$SCRIPTPATH/CalDAVTester"
-PYTHONPATH="$SCRIPTPATH/pycalendar/src" python testcaldav.py --print-details-onfail --basedir "$SCRIPTPATH/../caldavtest/" -o cdt.txt \
+PYTHONPATH="$SCRIPTPATH/pycalendar/src" python2 testcaldav.py --print-details-onfail --basedir "$SCRIPTPATH/../caldavtest/" -o cdt.txt \
 	"CalDAV/current-user-principal.xml" \
 	"CalDAV/sync-report.xml"
 

--- a/apps/dav/tests/ci/caldav/script.sh
+++ b/apps/dav/tests/ci/caldav/script.sh
@@ -9,7 +9,7 @@ sleep 30
 
 # run the tests
 cd "$SCRIPTPATH/CalDAVTester"
-PYTHONPATH="$SCRIPTPATH/pycalendar/src" python testcaldav.py --print-details-onfail --basedir "$SCRIPTPATH/../caldavtest/" -o cdt.txt \
+PYTHONPATH="$SCRIPTPATH/pycalendar/src" python2 testcaldav.py --print-details-onfail --basedir "$SCRIPTPATH/../caldavtest/" -o cdt.txt \
 	"CalDAV/current-user-principal.xml" \
 	"CalDAV/sync-report.xml" \
 	"CalDAV/sharing-calendars.xml"

--- a/apps/dav/tests/ci/carddav-old-endpoint/script.sh
+++ b/apps/dav/tests/ci/carddav-old-endpoint/script.sh
@@ -13,7 +13,7 @@ sleep 30
 
 # run the tests
 cd "$SCRIPTPATH/CalDAVTester"
-PYTHONPATH="$SCRIPTPATH/pycalendar/src" python testcaldav.py --print-details-onfail --basedir "$SCRIPTPATH/../caldavtest/" -o cdt.txt \
+PYTHONPATH="$SCRIPTPATH/pycalendar/src" python2 testcaldav.py --print-details-onfail --basedir "$SCRIPTPATH/../caldavtest/" -o cdt.txt \
 	"CardDAV/current-user-principal.xml" \
 	"CardDAV/sync-report.xml"
 

--- a/apps/dav/tests/ci/carddav/script.sh
+++ b/apps/dav/tests/ci/carddav/script.sh
@@ -9,7 +9,7 @@ sleep 30
 
 # run the tests
 cd "$SCRIPTPATH/CalDAVTester"
-PYTHONPATH="$SCRIPTPATH/pycalendar/src" python testcaldav.py --print-details-onfail --basedir "$SCRIPTPATH/../caldavtest/" -o cdt.txt \
+PYTHONPATH="$SCRIPTPATH/pycalendar/src" python2 testcaldav.py --print-details-onfail --basedir "$SCRIPTPATH/../caldavtest/" -o cdt.txt \
 	"CardDAV/current-user-principal.xml" \
 	"CardDAV/sync-report.xml" \
 	"CardDAV/sharing-addressbooks.xml"

